test/test-pages/mozilla-1/expected.html

```diff
@@ -2,65 +2,76 @@
     <div role="main" id="main-content">
         <section id="intro">
             <div>
-                <p>It’s easier than ever to personalize Firefox and make it work the way you do. <br/>No other browser gives you so much choice and flexibility.</p>
+                <p>It’s easier than ever to personalize Firefox and make it work the way you do. <br />No other browser gives you so much choice and flexibility.</p>
                 <div id="flexible-top-animation">
-                    <p><img src="http://mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/animations/flexible-top-fallback.c960365ba781.png" alt="" /> </p>
+                    <p><img src="http://mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/animations/flexible-top-fallback.c960365ba781.png" alt="" /></p>
                 </div>
             </div>
         </section>
         <section id="designed" data-ga-label="Designed to be redesigned">
             <div>
                 <div id="designed-copy">
-                    <h2>Designed to <br/>be redesigned</h2>
+                    <h2>Designed to <br />be redesigned</h2>
                     <p>Get fast and easy access to the features you use most in the new menu. Open the “Customize” panel to add, move or remove any button you want. Keep your favorite features — add-ons, private browsing, Sync and more — one quick click away.</p>
-                    <p><img src="http://mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/designed-redesigned-high-res.6efd60766484.png" data-processed="false" data-src="//mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/designed-redesigned.fbd3ee9402e6.png" data-high-res="true" data-high-res-src="//mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/designed-redesigned-high-res.6efd60766484.png" alt="" id="designed-mobile" /> </p>
+                    <p><img src="http://mozorg.cdn.mozilla.net/media/img/firefox/desktop/customize/designed-redesigned.fbd3ee9402e6.png" alt="" id="designed-mobile" />
```

Comment:
So it's hard to know what would happen here today, because the original images no longer load and the page has disappeared off google, but the file names here indicate that we're now loading the mobile version of the image instead of the high-res one. That's the opposite of what you wanted to accomplish, right?

I'm not sure there's a way to really solve both cases, short of trying to come up with heuristics to determine which URL is better - or creating some kind of pluggability where the consumer of the library can fetch both and determine which they want... which seems tricky, because it more or less entails starting to use `await` and promises in the library, which is its own project.

Could we keep the old/other source available on the image in a separate attribute, so that consumers of the resulting markup could still make their own decisions later? That might be an easier way of giving control to the library consumers without needing too many architectural changes inside readability.

Reply:
Done in [d32f39f ](https://github.com/mozilla/readability/pull/588/commits/d32f39f6c6e73c4edff91fb958b4faf4c549e5ff)

Reply:
I'm a bit puzzled because the image for which this conversation was started still only has 1 attribute?

Reply:
Looks like this conflicts with the code from #542 in `_fixLazyImages`, because your patch is replacing the original image with the one from inside the `noscript` element, without preserving the `data-src` etc. attributes.

I'll add a suggestion on how to fix this.